### PR TITLE
📽️ Fix values of pre_process_data_config for text generation

### PR DESCRIPTION
**Pull request by trajepl**
## Describe your changes

- make ["text"] default value for text_cols and update all example json config
- `corpus_strategy` default value: `join`
- `jointer` default value: `\n\n`
- `source_max_len` default value: 1024 https://github.com/artidoro/qlora/blob/7f4e95a68dc076bea9b3a413d2b512eca6d004e5/qlora.py#L111

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [ ] Is this PR including examples changes? If yes, please remember to update [example documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

### CODE REVIEW

**Comment by trajepl** [comment]
@jambayk  Could we determine the default value for `corpus_strategy` based on the task and/or dataset? I set `join` as default value.

File: examples/phi3/pass_configs/data_configs.json [hunk 1]
```diff
@@ -14,9 +14,7 @@
         },
         "pre_process_data_config": {
             "params": {
-                "corpus_strategy": "join",
                 "text_template": "### Question: {prompt} \n### Answer: {response}",
```
Comment (shaahji) [inline examples/phi3/pass_configs/data_configs.json:17]: The terminating ',' at the end of the line makes this a non-compliant json.

File: examples/llama2/notebook/llama2/config.json [hunk 2]
```diff
@@ -121,10 +121,7 @@
             },
             "pre_process_data_config": {
                 "params": {
-                    "dataset_type": "corpus",
-                    "corpus_strategy": "join",
                     "text_template": "### Question: {prompt} \n### Answer: {response}",
```
Comment (shaahji) [inline examples/llama2/notebook/llama2/config.json:124]: Extra comma at end-of-line here too.

**Review by shaahji — DISMISSED** [review]

**Comment by jambayk** [comment]
> @jambayk Could we determine the default value for `corpus_strategy` based on the task and/or dataset? I set `join` as default value.

I don't think that's possible. It depends on the type of preprocessing the user wants to do so it's not dependent on dataset. And the task for this is always text generation.

File: examples/phi3/pass_configs/data_configs.json (hunk 1 above)
Comment (trajepl) [reply examples/phi3/pass_configs/data_configs.json]: Removed.

File: examples/llama2/notebook/llama2/config.json (hunk 2 above)
Comment (trajepl) [reply examples/llama2/notebook/llama2/config.json]: Removed.

**Review by jambayk — APPROVED** [review]

**Pull request merged**